src/terrain/routes/tags.clj

```diff
@@ -10,32 +13,76 @@
   (util/optional-routes
    [#(and (config/filesystem-routes-enabled) (config/metadata-routes-enabled))]
 
-   (GET "/filesystem/entry/tags" []
-     (tags/list-all-attached-tags))
+   (context "/filesystem/entry" []
+            :tags ["tags"]
 
-   (DELETE "/filesystem/entry/tags" []
-     (tags/remove-all-attached-tags))
+      (GET "/tags" []
+        :summary schema/GetTagsSummary
+        :description schema/GetTagsDescription
+        :return schema/AttachedTagsListing
+        (ok (tags/list-all-attached-tags)))
 
-   (GET "/filesystem/entry/:entry-id/tags" [entry-id]
-     (tags/list-attached-tags entry-id))
+      (DELETE "/tags" []
+        :summary schema/DeleteTagsSummary
+        :description schema/DeleteTagsDescription
+        (tags/remove-all-attached-tags)
+        (ok))
 
-   (PATCH "/filesystem/entry/:entry-id/tags" [entry-id type :as {body :body}]
-     (tags/handle-patch-file-tags entry-id type body))
+      (GET "/:entry-id/tags" []
+        :path-params [entry-id :- TargetIdParam]
+        :summary schema/GetAttachedTagSummary
+        :description schema/GetAttachedTagDescription
+        :return schema/TagList
+        (ok (tags/list-attached-tags entry-id)))
```

Comment:
Adding a context here for the `/:entry-id` or even the `/:entry-id/path` URL paths would eliminate the need for the duplicate `:path-params` entries. This code has a good example of that: https://github.com/cyverse-de/terrain/blob/eff4049c1b4ecbee7659ce1d2dc03633c8ebcc2f/src/terrain/routes/analyses.clj#L87-L95

Reply:
OK, cool - this is a great example.  I wasn't sure how far to go with contexts, like maybe just one level was more than enough, but this looks more useful.  Thanks!